obspy/io/hypodd/pha.py

```diff
@@ -0,0 +1,128 @@
+        pick = Pick(waveform_id=_waveform_id, phase_hint=phase,
+                    time=time + float(reltime))
+        arrival = Arrival(phase=phase, pick_id=pick.resource_id,
+                          time_weight=float(weight))
+        picks.append(pick)
+        arrivals.append(arrival)
+    qu = None if rms == '0.0' else OriginQuality(standard_error=float(rms))
+    origin = Origin(arrivals=arrivals,
+                    quality=qu,
+                    latitude=float(la),
+                    longitude=float(lo),
+                    depth=1000 * float(dp),
+                    time=time)
+    magnitude = Magnitude(mag=mg)
+    event = Event(resource_id=id_,
+                  picks=picks,
+                  origins=[origin],
+                  magnitudes=[magnitude],
+                  preferred_origin_id=origin.resource_id,
+                  preferred_magnitude_id=magnitude.resource_id)
+    return event
+
+
+def _is_pha(filename):
+    try:
+        with open(filename, 'rb') as f:
+            temp = f.readline()
+    except Exception:
+        return False
+    return temp.startswith(b'#') and len(temp.split()) == 15
+
+
+def _read_pha(filename, inventory=None, id_map=None, id_default='.{}..{}',
+              ph2comp={'P': 'Z', 'S': 'N'}, encoding='utf-8'):
+    """
+    Read a HypoDD PHA file and returns an ObsPy Catalog object.
+
+    .. warning::
+        This function should NOT be called directly, it registers via the
+        ObsPy :func:`~obspy.core.event.read_events` function, call this
+        instead.
+
+    The optional parameters all deal with the problem, that the PHA format
+    only stores station names for the picks, but the Pick object expects
+    a SEED id.
+
+    :param str filename: File or file-like object in text mode.
+    :type inventory: :class:`~obspy.core.inventory.inventory.Inventory`
+    :param inventory: Inventory used to retrieve network code, location code
+        and channel code of stations (SEED id).
+    :param dict id_map: If channel information was not found in inventory,
+        it will be looked up in this dictionary
+        (example: `id_map={'MOX': 'GR.{}..HH{}'`).
+        The values must contain three dots and two `{}` which are
+        substituted by station code and component.
+    :param str id_default: Default SEED id expression.
+        The value must contain three dots and two `{}` which are
+        substituted by station code and component.
+    :param dict ph2comp: mapping of phases to components
+        (default: {'P': 'Z', 'S': 'N'})
+    :param str encoding: encoding used (default: utf-8)
+
+    :rtype: :class:`~obspy.core.event.Catalog`
+    :return: An ObsPy Catalog object.
+    """
+    seed_map = _seed_id_map(inventory, id_map, id_default)
+    with io.open(filename, 'r', encoding=encoding) as f:
+        text = f.read()
+    events = [_block2event(block, seed_map, id_default, ph2comp)
+              for block in text.split('#')[1:]]
```

Comment:
Such a fragile format - but if it always originates from the same code I guess this is reliable? Another option would be group the blocks with some regular expression but that might not be worth the effort here.

Reply:
I would leave it simple.